youtube_dl/extractor/yospace.py

```diff
@@ -0,0 +1,202 @@
+                    js_vars = re.findall(r'^\s*Reuters\.([^\s\[\]\.]+\.[^\s\[\]\.]+)\s*=\s*[\'"]?(.*?)[\'"]?;\s*\n', innerhtml, re.M)
+                    for ent in js_vars:
+                        if not ent[1]:
+                            continue
+                        if re.search(r'["\'].+?[\(\)\+]', ent[1]):
+                            continue
+                        rdata[ent[0]] = ent[1]
+                drawplayer_js = re.search(r'Reuters.yovideo.drawPlayer\((\{.+?\})\);', innerhtml, re.DOTALL)
+                if drawplayer_js:
+                    vidnum += 1
+                    ds = drawplayer_js.group(1)
+                    ds = re.sub(r':\s*Reuters\.([a-zA-Z_]+\.[a-zA-Z_]+)\s*,\s*\n', msub, ds)
+                    # "stuff_with": "variables like "+this("breaks")+" js_to_json";
+                    ds = re.sub(r'[\'"].+?[\'"]\s*:\s*\(.+,\s*\n', '', ds)
+                    vdata = self._parse_json(ds, 'parsed javascript chunk', transform_source=js_to_json)
+                    vdata['vidnum'] = vidnum
+                    ret.append(vdata)
+        return ret
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('id')
+        ret = []
+        webpage = self._download_webpage(url, video_id)
+        vids = self._scrape_javascript(webpage)
+        for vid in vids:
+            vurl = vid.get('flv', vid.get('mpeg'))
+            if vurl:
+                formats = []
+                formats.append({
+                    'url': vurl,
+                    'format_id': 'embed-flv-' + str(vid.get('vidnum')),
+                    'protocol': vurl.split(':')[0],
+                    'width': int_or_none(vid.get('width')),
+                    'height': int_or_none(vid.get('height')),
+                    'ext': 'flv',
+                    'tbr': 1080.0 if vid.get('vbc', 'vbcValue') == 'vbcValue' else float_or_none(vid.get('vbc')),
+                })
+                yo_id_str = re.search(r'yospace.+/(\d+)\?f=(\d+)', vurl)
+                if yo_id_str:
+                    yo_id = yo_id_str.group(1) + '/' + yo_id_str.group(2)
+                    murl = 'http://mas-e.cds1.yospace.com/mas/' + yo_id + '?trans=json'
+                    # yurl = 'http://csm-e.cds1.yospace.com/csm/'+yo_id
+                    formats.extend(self._extract_formats(murl, video_id))
+                if formats:
+                    self._sort_formats(formats)
+                    ret.append({
+                        'id': vid.get('id', video_id),
+                        'title': vid.get('title', video_id),
+                        'description': vid.get('description'),
+                        'webpage_url': url,
+                        'formats': formats,
+                    })
+        if not ret:
+            raise ExtractorError('No video found', expected=True)
+        if len(ret) > 1:
```

Comment:
I would always return a playlist.

Please add a test with more than one video, like [in the ted extractor](https://github.com/rg3/youtube-dl/blob/6621ca39a3f03e44648b54bd8ff57814362f5dbc/youtube_dl/extractor/ted.py#L64-L69).